### PR TITLE
Detect leaked services

### PR DESCRIPTION
**Pull request by oldergod**
Services are considered leaked if they have been GC'd before they were closed.

### CODE REVIEW

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt [hunk 1]
```diff
@@ -33,6 +33,7 @@ internal interface OutboundBridge {
     @PublishedApi internal val endpoint: Endpoint,
   ) {
     val serializersModule = json.serializersModule
+    var closed = false
```
Comment (oldergod) [inline zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt:36]: We don't define that one in IR, that's cool?

File: zipline-kotlin-plugin-hosted/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt [hunk 2]
```diff
@@ -763,16 +765,32 @@ internal class AdapterGenerator(
...
+        origin = IrDeclarationOrigin.DEFINED
+      }
+
+      val newCall = irCall(ziplineApis.outboundBridgeContextNewCall).apply {
+        dispatchReceiver = irGet(contextLocal)
         putValueArgument(0, irString(bridgedFunction.signature))
         putValueArgument(1, irInt(bridgedFunction.valueParameters.size))
       }
 
+      // If this is the close() function, tell the OutboundContext that this instance is closed.
```
Comment (swankjesse) [inline zipline-kotlin-plugin-hosted/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt:786]: Can you add a comment that matches what the source code would do?
```
// context.closed = true
```
I find it really helpful in IR stuff.

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt (hunk 1 above)
Comment (swankjesse) [reply zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt:36]: I don’t follow?

**Review by swankjesse — APPROVED** [review]

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt (hunk 1 above)
Comment (oldergod) [reply zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt:36]: We added code in IR to do `context.close = true`, don't we need IR code that add the line `var closed = false` ?

File: zipline-kotlin-plugin-hosted/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt (hunk 2 above)
Comment (oldergod) [reply zipline-kotlin-plugin-hosted/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt:786]: done

**Pull request merged**